microsoft/testsuites/core/timesync.py

```diff
@@ -377,3 +380,55 @@ def verify_pmu_disabled_for_arm64(self, node: Node) -> None:
         result = cat.run("/proc/cmdline", force_run=True)
         if "initcall_blacklist=arm_pmu_acpi_init" not in result.stdout:
             raise LisaException("PMU is not disabled in kernel cmdline")
+
+    @TestCaseMetadata(
+        description="""
+        This test is to verify that timedrift is automatically corrected by chrony
+        after a time jump.
+
+        Steps:
+        1. Set makestep to 1.0 -1 to allow Chrony to make large adjustments.
+        Ref: https://learn.microsoft.com/en-us/azure/virtual-machines/linux/time-sync#chrony # noqa: E501
+        2. Manually change the system clock to a time in the past.
+        3. Verify that Chrony has corrected the time drift.
+        """,
+        priority=1,
+    )
+    def verify_timedrift_corrected(self, node: Node, log: Logger) -> None:
+        # Initialize chrony
+        chrony = node.tools[Chrony]
+        chrony.set_makestep("1.0 -1")
+        chrony.restart()
+
+        for iteration in range(3):
+            is_drift_corrected = False
+
+            # Get current time
+            date = node.tools[Date]
+            node_time = date.current()
+            modified_time = deepcopy(node_time)
+            modified_time = modified_time.replace(year=node_time.year - 2)
+            date.set(modified_time)
+
+            # Poll every second and check if the time drift is corrected
+            # for 5 minutes
+            for _ in range(300):
+                time.sleep(1)
+                current_time = date.current()
+                if (
+                    current_time.year == node_time.year
```

Comment:
There maybe a bug. if the test case running at last minute a day, after 5 minutes, the day maybe not the same anyway. The year and month have the same problem. Please find another stable way to check it.

Reply:
Fixed